src/java/org/apache/cassandra/schema/SchemaKeyspace.java

```diff
@@ -568,8 +568,7 @@ private static void addTableParamsToRowBuilder(TableParams params, Row.SimpleBui
 
         // As above, only add the allow_auto_snapshot column if the value is not default (true) and
         // auto-snapshotting is enabled, to avoid RTE in pre-4.2 versioned node during upgrades
-        if (DatabaseDescriptor.isAutoSnapshot() && !params.allowAutoSnapshot)
-            builder.add("allow_auto_snapshot", false);
+        builder.add("allow_auto_snapshot", params.allowAutoSnapshot);
```

Comment:
You still want to gate it on `!params.allowAutoSnapshot`, so it's only persisted if explicitly set to `false`. Just don't involve `DatabaseDescriptor.isAutoSnapshot()`.